docs/Project.toml

```diff
@@ -1,5 +1,4 @@
 [deps]
-AbstractAlgebra = "c3fe647b-3220-5bb0-a1ea-a7954cac585d"
```

Comment:
Any reason to remove this? I keep adding it back when I build the documentation locally, which is why I finally committed it. Note that the CI job building the docs also adds it (we just don't commit there, obviously).

Reply:
Oh wait, I misread this. What I mean is that I keep adding *Nemo* to `docs/Project.toml`... But I agree that AA should be removed here (with the other changes you made below this is possible)

Reply:
Somewhere in the julia docs I read that the package itself shouldn't be j it's docs/test project deps.
But yeah, this is something different

Reply:
Yeah I slowly start to remember, I've run into this before (probably more than once). As the error in the CI logs indicate, `using AbstractAlgebra` won't just silently proceed when there is already a module `AbstractAlgebra` present. Perhaps it works if you do `using Nemo.AbstractAlgebra`? But probably also not.... So either you need to hack the system to convince it that AA is already loaded. Or we change AA: Maybe we can change it to do `using ..AbstractAlgebra` in the individual `jldoctest` blocks, i.e.:
````
```jldoctest; setup = :(using ..AbstractAlgebra)
````
Empirical evidence why this might work: in a fresh Julia session:
```
julia> using AbstractAlgebra

julia> using ..AbstractAlgebra
```
In another fresh Julia session (with a different environment that only has Nemo but not AA):
```
julia> using Nemo

Welcome to Nemo version 0.37.1

Nemo comes with absolutely no warranty whatsoever


julia> using AbstractAlgebra
 │ Package AbstractAlgebra not found, but a package named AbstractAlgebra is available from a registry.
 │ Install package?
 │   (1.9) pkg> add AbstractAlgebra
 └ (y/n/o) [y]: n
ERROR: ArgumentError: Package AbstractAlgebra not found in current path.
- Run `import Pkg; Pkg.add("AbstractAlgebra")` to install the AbstractAlgebra package.
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1630 [inlined]
 [2] macro expansion
   @ ./lock.jl:267 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1611

julia> AbstractAlgebra
ERROR: UndefVarError: `AbstractAlgebra` not defined

julia> using Nemo.AbstractAlgebra

julia> AbstractAlgebra
AbstractAlgebra

julia> using AbstractAlgebra
 │ Package AbstractAlgebra not found, but a package named AbstractAlgebra is available from a registry.
 │ Install package?
 │   (1.9) pkg> add AbstractAlgebra
 └ (y/n/o) [y]: n
ERROR: ArgumentError: Package AbstractAlgebra not found in current path, maybe you meant `import/using .AbstractAlgebra`.
- Otherwise, run `import Pkg; Pkg.add("AbstractAlgebra")` to install the AbstractAlgebra package.
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1630 [inlined]
 [2] macro expansion
   @ ./lock.jl:267 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1611

julia> using ..AbstractAlgebra

```


Of course that would require making an AA PR now and then releasing it

Reply:
> Somewhere in the julia docs I read that the package itself shouldn't be j it's docs/test project deps. But yeah, this is something different

It would be interesting to know where you read this, and how whoever wrote it then builds their docs? Looking at [the Documenter.jl documentation](julia --project make.jl) they suggest to build package docs via
```
cd docs
julia --color=yes --project make.jl
```
Which of course only works if you just before did something like this:
```
julia --color=yes --project -e 'using Pkg ; Pkg.develop(path="..")'
```
which is precisely what I do (and essentially also what our CI does), except that then I have a diff in my Project.toml...

Reply:
I really do not like that we need to change AA for this. Thus I would just re-add it to `docs/Project.toml` and deal with it sometime later.